src/gc/impl/conservative/gc.d

```diff
@@ -252,6 +253,10 @@ debug (LOGGING)
 
 class ConservativeGC : GC
 {
+    bool profiling_enabled;
+    shared Mutex profiler_lock;
```

Comment:
Why creating this new mutex instead of using the pre-existing GC global lock for everything?